samples/javascript_nodejs/46.teams/bots/dialogBot.js

```diff
@@ -0,0 +1,49 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+const { TeamsActivityHandler } = require('../teamsActivityHandler');
+
+class DialogBot extends TeamsActivityHandler {
+    /**
+     *
+     * @param {ConversationState} conversationState
+     * @param {UserState} userState
+     * @param {Dialog} dialog
+     * @param {any} logger object for logging events, defaults to console if none is provided
+     */
+    constructor(conversationState, userState, dialog, logger) {
+        super();
+        if (!conversationState) throw new Error('[DialogBot]: Missing parameter. conversationState is required');
+        if (!userState) throw new Error('[DialogBot]: Missing parameter. userState is required');
+        if (!dialog) throw new Error('[DialogBot]: Missing parameter. dialog is required');
+        if (!logger) {
+            logger = console;
+            logger.log('[DialogBot]: logger not passed in, defaulting to console');
+        }
+
+        this.conversationState = conversationState;
+        this.userState = userState;
+        this.dialog = dialog;
+        this.logger = logger;
```

Comment:
@johnataylor can you mirror the changes from #1606 and remove the `logger` from this new sample?